### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by BlauwaLiu**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -55,12 +55,12 @@ build-iPhoneSimulator/
 # Used by RuboCop. Remote config files pulled in from inherit_from directive.
 # .rubocop-https?--*
 
-# SOA-specific
-config/secrets.yml
```
Comment (chinyakao) [inline .gitignore:59]: IMPORTANT
`secretes.yml` should not upload to version control
and keep this line in gitignore.

I suggest you provide a `secrets_example.yml `with necessary real key and fake key. refers to professor's file. [proffersor's file](https://github.com/ISS-SOA/soa2021-codepraise/blob/main/config/secrets_example.yml)

File: app/models/repositories/firms.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      #   # SELECT * FROM `projects` LEFT JOIN `members`
+      #   # ON (`members`.`id` = `projects`.`owner_id`)
+      #   # WHERE ((`username` = 'owner_name') AND (`name` = 'project_name'))
+      #   db_project = Database::FirmOrm
+      #     .left_join(:members, id: :owner_id)
+      #     .where(username: owner_name, name: project_name)
+      #     .first
+      #   rebuild_entity(db_project)
+      # end
+
```
Comment (chinyakao) [inline app/models/repositories/firms.rb:23]: the useless code should not upload to version control even you comment it.

File: config/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (chinyakao) [inline config/secrets.yml:1]: This file should be `secrets_example.yml`

**Review by chinyakao — COMMENTED** [review]
review week8

**Comment by chinyakao** [comment]
@BlauwaLiu @schwarz-tw @NianHung please take a look

File: .gitignore (hunk 1 above)
Comment (schwarz-tw) [reply .gitignore:59]: We did not put out real credentials in the `secrets.yml`, so it seem that we just need to rename it to `secrets_example.yml`, thank you for reminding us!

File: app/models/repositories/firms.rb (hunk 2 above)
Comment (schwarz-tw) [reply app/models/repositories/firms.rb:23]: copy that!

**Pull request merged**